library/data/tags/src/main/java/org/quiltmc/qsl/tag/impl/client/ClientTagRegistryManager.java

```diff
@@ -33,60 +39,82 @@
 
 import net.minecraft.resource.ResourceManager;
 import net.minecraft.tag.Tag;
-import net.minecraft.tag.TagGroup;
 import net.minecraft.tag.TagGroupLoader;
+import net.minecraft.tag.TagKey;
+import net.minecraft.tag.TagManagerLoader;
+import net.minecraft.util.Holder;
 import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.BuiltinRegistries;
 import net.minecraft.util.registry.DynamicRegistryManager;
 import net.minecraft.util.registry.Registry;
 import net.minecraft.util.registry.RegistryKey;
 
+import org.quiltmc.qsl.tag.api.QuiltTagKey;
+import org.quiltmc.qsl.tag.api.TagRegistry;
 import org.quiltmc.qsl.tag.api.TagType;
-import org.quiltmc.qsl.tag.impl.TagDelegate;
 import org.quiltmc.qsl.tag.impl.TagRegistryImpl;
-import org.quiltmc.qsl.tag.mixin.RequiredTagListRegistryAccessor;
 import org.quiltmc.qsl.tag.mixin.client.TagGroupLoaderAccessor;
 
 @ApiStatus.Internal
 public final class ClientTagRegistryManager<T> {
 	private static final Map<RegistryKey<? extends Registry<?>>, ClientTagRegistryManager<?>> TAG_GROUP_MANAGERS =
-			new Object2ObjectOpenHashMap<>();
+			new WeakHashMap<>();
```

Comment:
The RegistryKey itself!

The reason I made it weak is because TagKeys are usually made *before* and they usually stay in the global memory since it's very often public static final fields, which also keep the RegistryKey instance alive.